Sources/SnapshotPreviewsCore/ViewInspection.swift

```diff
@@ -67,25 +67,45 @@ enum ViewInspection {
     return [(view, [])]
   }
 
-  public static func preferredColorScheme(of view: some View) -> ColorScheme? {
+  private static func getModifier<T>(of view: some View, parseModifier: (Any) -> T?) -> T? {
     let typeName = String(reflecting: type(of: view))
     if typeName.starts(with: "SwiftUI.ModifiedContent") {
       let modifier = Self.attribute(label: "modifier", value: view)!
-      if let colorSchemePreference = modifier as? _PreferenceWritingModifier<PreferredColorSchemeKey> {
-        return colorSchemePreference.value
-      } else if let colorSchemePreference = modifier as? _PreferenceWritingModifier<PreferredColorSchemeKey> {
-        return colorSchemePreference.value
-      } else {
-        let content = Self.attribute(label: "content", value: view) as! any View
-        return preferredColorScheme(of: content)
+      if let result = parseModifier(modifier) {
+        return result
       }
+      let content = Self.attribute(label: "content", value: view) as! any View
+      return getModifier(of: content, parseModifier: parseModifier)
     } else if typeName.starts(with: "SwiftUI.AnyView") {
       let storage = Self.attribute(label: "storage", value: view)!
       let storageView = Self.attribute(label: "view", value: storage) as! any View
-      return preferredColorScheme(of: storageView)
+      return getModifier(of: storageView, parseModifier: parseModifier)
     }
     return nil
   }
+
+  public static func preferredColorScheme(of view: some View) -> ColorScheme? {
+    return getModifier(of: view) { modifier in
+      if let colorSchemePreference = modifier as? _PreferenceWritingModifier<PreferredColorSchemeKey> {
+        return colorSchemePreference.value
+      } else if let colorSchemePreference = modifier as? _PreferenceWritingModifier<PreferredColorSchemeKey> {
+        return colorSchemePreference.value
+      }
+      return nil
+    }
+  }
+
+  public static func precision(of view: some View) -> Float? {
+    return getModifier(of: view) { modifier in
+      let typeName = String(reflecting: modifier)
+      if typeName.contains(".PrecisionModifier") {
```

Comment:
nit: maybe we should prefix the type as well so that it shows up in reflection, like `EmergePrecisionModifier`. Unlikely to ever happen but `PrecisionModifier` sounds generic enough.